do-not-commit/listeneval.py

```diff
@@ -0,0 +1,9 @@
+from logging.config import listen
+
+PORT_NUMBER = 1234
+
+
+def start_log():
+    # ruleid: listen-eval
+    t = listen(PORT_NUMBER)
```

Comment:
Because portions of the logging configuration are passed through eval(),
use of this function may open its users to a security risk. While the
function only binds to a socket on localhost, and so does not accept
connections from remote machines, there are scenarios where untrusted
code could be run under the account of the process which calls listen().
See more details at https://docs.python.org/3/library/logging.config.html?highlight=security#logging.config.listen ([dit.logging.listeneval.listen-eval](https://semgrep.dev/r/dit.logging.listeneval.listen-eval))
#
<div>
<a href="https://semgrep.dev">
<img align="right" src="https://r2c.dev/semgrep_icon_text.svg" height=28px alt="" />
</a>

<sub> [How do I suppress this?](https://semgrep.dev/docs/running-rules/#managing-findings) **·** [Report a bug](https://github.com/returntocorp/semgrep-rules/issues/new?assignees=&labels=&template=rule_bug_report.md&title=dit.logging.listeneval.listen-eval) **·** [Manage rule](https://semgrep.dev/manage/projects)</sub>
</div>